runtime/src/iree/hal/local/loaders/system_library_loader.c

```diff
@@ -383,9 +383,23 @@ static iree_status_t iree_hal_system_executable_issue_call(
   if (iree_string_view_is_empty(entry_point_name)) {
     entry_point_name = iree_make_cstring_view("unknown_dylib_call");
   }
-  IREE_TRACE_ZONE_BEGIN_EXTERNAL(
-      z0, executable->identifier.data, executable->identifier.size, ordinal,
-      entry_point_name.data, entry_point_name.size, NULL, 0);
+  const char* source_file = NULL;
+  size_t source_file_length = 0;
+  uint32_t source_loc;
+  if (library->exports.src_files != NULL && library->exports.src_locs != NULL) {
+    // We have source location data, so use it.
+    source_file = library->exports.src_files[ordinal];
+    source_file_length = strlen(source_file);
```

Comment:
Unfortunate strlen as this is on the hot path (every dispatch workgroup).
To avoid the strlen we could instead store length-prefixed strings in the src loc table.

It's annoying too that we do this for export names and that'd be good to fix in the same way in the future (but they're usually shorter than file paths... until we start naming them all clever-like).